### PR TITLE
[CPU] Add MXFP4 W4A16 MoE support

### PR DESCRIPTION
**Pull request by yuwenzho**
## Purpose
Adapted from https://github.com/sgl-project/sglang/pull/16775
- Sync csrc/cpu/sgl-kernels/ with upstream SGLang sgl-kernel. 
- Add MXFP4 W4A16 fused MoE experts for CPU, targeting openai/gpt-oss-20b

## Test Plan
python -m pytest tests/kernels/moe/test_cpu_mxfp4_fused_moe.py -v

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by yuwenzho** [comment]
@bigPYJ1151 please review

File: vllm/model_executor/layers/fused_moe/experts/cpu_moe.py [hunk 1]
```diff
@@ -171,5 +172,167 @@ def apply(
...
+) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
+    """VNNI-prepack MXFP4 MoE weights and repack scales for CPU AMX kernel."""
+    num_experts = w13.size(0)
+    packed_w13_list = []
+    packed_w2_list = []
+    for i in range(num_experts):
+        packed_w13_list.append(torch.ops._C.convert_weight_packed(w13[i]))
+        packed_w2_list.append(torch.ops._C.convert_weight_packed(w2[i]))
+    packed_w13 = torch.stack(packed_w13_list)
+    packed_w2 = torch.stack(packed_w2_list)
```
Comment (bigPYJ1151) [inline vllm/model_executor/layers/fused_moe/experts/cpu_moe.py]: ```convert_weight_packed``` will process the expert dim internally, no need to for-each here.

**Review by bigPYJ1151 — COMMENTED** [review]
Thanks @yuwenzho Overall looks good :) 
Please also help to increase the timeout to 30m. Currently the tests take about 18m, close to the limit.
https://github.com/vllm-project/vllm/blob/main/.buildkite/hardware_tests/cpu.yaml#L73

**Review by bigPYJ1151 — APPROVED** [review]
